code/controllers/subsystems/records.dm

```diff
@@ -239,56 +237,46 @@
 			var/depDat
 			for(var/list/item in depI)
 				depDat += "<li><strong>[item["name"]]</strong> - [item["rank"]] ([item["active"]])</li>"
-			dat += "<h3>[nameMap[dep]]</h3><ul>[depDat]</ul>"
+			dat += "<h3>[dep]</h3><ul>[depDat]</ul>"
 	return dat
 
 /datum/controller/subsystem/records/proc/get_manifest_list()
 	if(manifest.len)
 		return manifest
-	manifest = list(
-		"heads" = list(),
-		"sec" = list(),
-		"eng" = list(),
-		"med" = list(),
-		"sci" = list(),
-		"car" = list(),
-		"civ" = list(),
-		"bot" = list(),
-		"misc" = list()
-	)
-	var/positions = list(
-		"heads" = command_positions,
-		"sec" = security_positions,
-		"eng" = engineering_positions,
-		"med" = medical_positions,
-		"sci" = science_positions,
-		"car" = cargo_positions,
-		"civ" = civilian_positions,
-		"bot" = nonhuman_positions
-	)
+	manifest = list()
+	for(var/department in positions_by_department)  // prepare empty manifest
+		manifest[department] = list()
 	for(var/datum/record/general/t in records)
 		var/name = sanitize(t.name, encode = FALSE)
 		var/rank = sanitize(t.rank, encode = FALSE)
 		var/real_rank = make_list_rank(t.real_rank)
 
 		var/isactive = t.physical_status
-		var/department = 0
-		var/depthead = 0            // Department Heads will be placed at the top of their lists.
-
-		for(var/positionType in positions)
-			var/typesPositions = positions[positionType]
-			if(real_rank in typesPositions)
-				manifest[positionType][++manifest[positionType].len] = list("name" = name, "rank" = rank, "active" = isactive)
-				department = 1
-				if ((depthead || rank == "Captain") && manifest[positionType].len != 1)
-					manifest[positionType].Swap(1, manifest[positionType].len)
-					manifest[positionType][1]["head"] = TRUE
-				if(positionType == "head")
-					depthead = 1
-
-		if(!department && !(name in manifest["heads"]))
-			manifest["misc"][++manifest["misc"].len] = list("name" = name, "rank" = rank, "active" = isactive)
+		var/dept = null
+		var/depthead = FALSE
+
+		for(var/department in positions_by_department)
+			if(!(real_rank in positions_by_department[department])) // search for a department with that rank
+				continue
+			dept = department
+			if(department == DEPARTMENT_COMMAND)
+				depthead = TRUE
+				continue
+			// because Command is first we have guarantee that the depthead var has been assigned and we have found the actual dept so we can safely break
+			break
+
+		if(isnull(dept)) // no department found
+			dept = DEPARTMENT_MISCELLANEOUS
+
+		 // add them to their department
+		manifest[dept][++manifest[dept].len] = list("name" = name, "rank" = rank, "active" = isactive, "head" = depthead)
+		if(dept == DEPARTMENT_COMMAND) // they are a captain, put them on top of command
```

Comment:
1. Isn't this also true if you're a department head?

2. It feels kind of weird to have `DEPARTMENT_COMMAND` mean two different things, since it is used as a key in `positions_by_department` and as a var on `/datum/job` - It doesn't mean the same thing for the Captain as it does for a Department Head. 

Since this line is supposed to uniquely identify the Captain job, it should be something like `rank == "Captain"` or some map-defined head-of-heads var that would be Captain.

Reply:
1. It is not, department heads have the department they are a head of as ... their department. Well technically it _is_ true but thanks to the code above (and the fact that Command is first) they get assigned Command first and then it's immediately overwritten with their actual department.
2. It _is_ weird but command is just special that way. It's not a real department since the only person actually assigned to it is the captain; everything else is "copies". I don't see a way out of this short of completely reworking how we think about Command and allowing people to _actually_ be in multiple departments at once while being a head in one and not the other.
3. I specifically left it this way to (1) not have to hardcode the Captain rank here and (2) if it ever happens that we have some proper member in there _other than captain_ (like a captain's / command secretary or something) it would get assigned in there properly.

Overall this is something I would prefer not to change; I had to touch it mostly because I added the constants and had to get rid of that weird "map".

Reply:
Also fixed by asking SSJobs for the job.